### PR TITLE
feat(client): create row component

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: tools/ui-components/src/row/row.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+import { RowProps } from './types';
+
+export const Row = ({ className, children }: RowProps) => {
+  if (!className) {
+    className = '';
+  }
+  return <div className={`mx-[-15px] ${className}`}>{children}</div>;
```
Comment (ojeytonwilliams) [inline tools/ui-components/src/row/row.tsx:9]:
```suggestion
  if (!className) {
    className = '';
  }
  return <div className={`mx-[-15px] ${className ?? ''}`}>{children}</div>;
```
Nitpick :man_shrugging:

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: tools/ui-components/src/row/row.tsx (hunk 1 above)
Comment (ahmaxed) [reply tools/ui-components/src/row/row.tsx:9]: there should be a linter for these 🤔

Comment (Sboonny) [reply tools/ui-components/src/row/row.tsx:9]: no linter rule please, if statement are easier to read than the other one

Comment (ojeytonwilliams) [reply tools/ui-components/src/row/row.tsx]: In isolation, arguably, but even then `classname = classname ?? ''` is clearer. However, the broader point is that it's unhelpful to force readers to backtrack when reading code. If I have to keep a mental stack, that's effort. If it's unnecessary, then we should fix it :man_shrugging:

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

File: tools/ui-components/src/row/row.tsx (hunk 1 above)
Comment (Sboonny) [reply tools/ui-components/src/row/row.tsx]: true when it's unnecessary, but linter role force us to deal with the necessary the wrong way.

Comment (ojeytonwilliams) [reply tools/ui-components/src/row/row.tsx]: Oh, yeah, I agree. It would be an impressive linter that could make this kind of subjective call!

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**